Fix x11 mouse scroll wheel

rukai opened the pull request:
Fixes: https://github.com/tomaka/winit/issues/217
*   Direction issue was resolved by swapping the order of the operands in `(unsafe { *value } - info.position)`
*   "Stored up" scrolls issue was resolved by updating the scroll position, stored in the relevant Device, every XI_Enter event

Ralith commented:
> Direction issue

Scroll direction seems correct to me--positive is down and right. What behavior are you observing? What do other platforms do? Bear in mind that some platforms (e.g. OSX) have a reversed direction convention.

> "Stored up" scrolls issue was resolved by updating the scroll position, stored in the relevant Device, every XI_Enter event

That part looks good, thanks.

rukai commented:
Well this is weird.

Found a file in /etc/X11/xorg.conf.d
```
Section "InputClass"
 Identifier "Joystick hat mapping"
 Option "StartKeysEnabled" "True"
 #MatchIsJoystick "on"
 Option "MapAxis5" "keylow=113 keyhigh=114"
 Option "MapAxis6" "keylow=111 keyhigh=116"
EndSection
```
I removed the file and rebooted.
The scroll direction was fixed!

Then I readded the file and rebooted.
Issue still fixed! wat.

The file has probably been there for years. (I dont actually know why I added it or what it was for)
I had previously checked out commit 15aafc2908465448b4e7fc596f858d9ee4f4765b and the previous commit and verified that was when the issue was introduced.

I guess its just an X11 bug that didnt occur in the previous way winit used the API.
And the bug was reset by clearing some sort of cache when the configuration was altered.

rukai commented:
I have removed the incorrect flipping of direction.

The scroll position update should be good to merge.

Ralith commented on src/platform/linux/x11/mod.rs:
```diff
@@ -406,7 +406,7 @@ impl EventsLoop {
                                 if ffi::XIMaskIsSet(mask, i) {
                                     if let Some(&mut (_, ref mut info)) = physical_device.scroll_axes.iter_mut().find(|&&mut (axis, _)| axis == i) {
-                                        let delta = (unsafe { *value } - info.position) / info.increment;
+                                        let delta = (unsafe { *value } - info.position) / info.increment; // TODO: Doesn't handle underflow/overflow
```
Is overflow a realistic concern here? The range of double is astronomical, and we're only computing differences between successive values delivered by X.

rukai commented:
Didnt pay attention to the type, good point!
Fixed.

Ralith commented on Cargo.toml:
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "winit"
-version = "0.7.2"
+version = "0.7.3"
```
I think bumping this is usually left for the maintainer to do immediately prior to making a release.

tomaka commented:
I don't mind the version bump.
